.github/workflows/pull_request.workflow.yml

```diff
@@ -0,0 +1,156 @@
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/es-lint
+
+  unit-tests:
+    name: Unit Tests - Node.js v${{ matrix.node-version }}
+    needs: ['lint']
+    strategy:
+      matrix:
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/unit-tests
+
+  functional-tests-legacy:
+    name: Functional Tests Legacy - ${{ matrix.test_set }} - Node.js v${{ matrix.node-version }}
+    needs: ['unit-tests']
+    strategy:
+      matrix:
+        test_set: ['http', 'websocket', 'mqtt']
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/functional-tests
+        env:
+          KUZZLE_FUNCTIONAL_TESTS: test:functional:legacy:${{ matrix.test_set }}
+        with:
+          docker-compose-test-file: ./.ci/test.yml
+
+  functional-tests-sdk:
+    name: Functional Tests SDK - ${{ matrix.test_set }} - Node.js v${{ matrix.node-version }}
+    needs: ['unit-tests']
+    strategy:
+      matrix:
+        test_set: ['http', 'websocket']
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/functional-tests
+        env:
```

Comment:
This env var should be put inside the actions related to it and given by the with: property